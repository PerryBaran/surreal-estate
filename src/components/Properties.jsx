import React, { useEffect, useState } from "react";
import style from "../styles/properties.module.css";
import getProperty from "../requests/getProperty";
import PropertyCard from "./PropertyCard";
import Alert from "./Alert";

const Properties = () => {
  const [properties, setProperties] = useState([]);
  const [alert, setAlert] = useState("");

  useEffect(() => {
    getProperty(setProperties, setAlert);
  }, []);

  return (
    <div className={style.properties}>
      <Alert message={alert} />
      <div className={style["properties-array"]}>
        {properties.map((property) => {
          return <PropertyCard key={property._id} {...property} />;
        })}
      </div>
    </div>
  );
};

export default Properties;
