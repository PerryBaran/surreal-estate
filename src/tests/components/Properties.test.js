import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import axios from "axios";
import Properties from "../../components/Properties";

describe("Properties", () => {
  const mockResponse = {
    data: [
      {
        title: "title 1",
        type: "type 1",
        bathrooms: "1",
        bedrooms: "1",
        price: "1000",
        city: "city 1",
        email: "email@email.com",
      },
      {
        title: "titile 2",
        type: "type 2",
        bathrooms: "2",
        bedrooms: "2",
        price: "2000",
        city: "city 2",
        email: "fake@email.com",
      },
    ],
  };

  beforeEach(() => {
    jest.spyOn(axios, "get").mockResolvedValue(mockResponse);
  });

  test("snapshot", () => {
    const { asFragment } = render(<Properties />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders properties", async () => {
    render(<Properties />);

    await waitFor(() =>
      expect(screen.getAllByAltText("property")).toHaveLength(
        mockResponse.data.length
      )
    );
  });
});
