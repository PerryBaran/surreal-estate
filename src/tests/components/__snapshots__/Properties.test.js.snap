// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties snapshot 1`] = `
<DocumentFragment>
  <div
    class="properties"
  >
    <div
      class="properties-array"
    />
  </div>
</DocumentFragment>
`;
